File: backend/gameHubBack/services/BatalhaRural/Rules/BoundRuler.cs
namespace gameHubBack.Services.BatalhaRural.Rules;

public static class BoundRuler
{
    public static bool IsOutOfBounds(TokenInfo tokenInfo, string[][] tiles)
    {
        if (tokenInfo.BaseZeroFinalPosX >= tiles.Length || tokenInfo.BaseZeroFinalPosY >= tiles.Length) return true;

        if (tokenInfo.BaseZeroFinalPosX < 0 || tokenInfo.BaseZeroFinalPosY < 0) return true;

        return false;
    }
}

File: frontend/GameHubFront/src/Games/batalhaRural/services/rules/BoundRuler.ts
import { Board } from "../../entities/Board";
import { Logger } from "../../helpers/Logger";
import { TokenInfo } from "../../helpers/TokenInfo";

export class BoundRuler {

    static isOutOfBounds(tokenInfo: TokenInfo, board: Board): boolean {
        if ((tokenInfo.baseZeroFinalPosX >= board.tiles.length) || (tokenInfo.baseZeroFinalPosY >= board.tiles.length)) {
            Logger.outOfBounds(tokenInfo)
            return true;
        }

        if ((tokenInfo.baseZeroFinalPosX < 0) || (tokenInfo.baseZeroFinalPosY < 0)) {
            Logger.outOfBounds(tokenInfo);
            return true;
        }

        return false;
    }
}